lib/craft/machine.ex

```diff
@@ -143,6 +148,15 @@ defmodule Craft.Machine do
     |> GenServer.call(request, timeout)
   end
 
+  @doc "Allows responding to a query from a different process, like GenServer.reply"
+  def reply({:direct, query_from}, reply) do
+    GenServer.reply(query_from, reply)
+  end
+
+  def reply({{:quorum, machine_pid}, query_from}, reply) do
```

Comment:
```suggestion
  def reply({:quorum, machine_pid, query_from}, reply) do
```

no need to embed the tuple, since we never match on just the first element, a flat "record" tuple is fine